src/platform/windows.rs

```diff
@@ -118,3 +121,55 @@ impl DeviceIdExtWindows for DeviceId {
         self.0.get_persistent_identifier()
     }
 }
+
+/// Unprocessed window event that are specific to Windows.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct OsSpecificWindowEvent {
+    pub(crate) window: HWND,
+    /// The message identifier of the window event (`WM_*`).
+    pub message: UINT,
+    /// The first parameter of the window event (`WPARAM`).
+    pub wparam: WPARAM,
+    /// The second parameter of the window event(`LPARAM`).
+    pub lparam: LPARAM,
+    pub(crate) retval: *mut Option<LRESULT>,
+}
+
+impl OsSpecificWindowEvent {
+    /// Marks the current event processed, returning the specified value
+    /// as the result of the window procedure to the system.
+    /// How the value is interpreted is dependent on the message's type.
+    /// The next window procedure (or subclass window
+    /// procedure) won't be called for this message.
+    ///
+    /// Calling this function outside this event's call to the closure passed to
+    /// [`EventLoop::run`](../../event_loop/struct.EventLoop.html#method.run)
+    /// will result in undefined behavior.
+    #[inline]
+    pub unsafe fn set_reply(&self, val: isize) {
+        *self.retval = Some(val);
+    }
+
+    /// Immediately calls the next window procedure(or subclass window
+    /// procedure), and maps its return value through the user-passed closure.
+    /// How the value is interpreted is dependent on the message's type.
+    ///
+    /// This function must not be called more than once for each message.
+    ///
```

Comment:
Can we mention that improperly messing with the params or messages could result in UB?

Reply:
Sure, will add it this evening(in my timezone)

Reply:
done.

Reply:
done